storage/influxdb/influxdb.go

```diff
@@ -229,28 +273,107 @@ func newStorage(
 	isSecure bool,
 	bufferDuration time.Duration,
 ) (*influxdbStorage, error) {
-	config := &influxdb.ClientConfig{
-		Host:     influxdbHost,
-		Username: username,
-		Password: password,
-		Database: database,
-		IsSecure: isSecure,
+	url := &url.URL{
+		Scheme: "http",
+		Host:   influxdbHost,
+	}
+	if isSecure {
+		url.Scheme = "https"
+	}
+
+	config := &influxdb.Config{
+		URL:       *url,
+		Username:  username,
+		Password:  password,
+		UserAgent: fmt.Sprintf("%v/%v", "cAdvisor", version.Info["version"]),
 	}
-	client, err := influxdb.NewClient(config)
+	client, err := influxdb.NewClient(*config)
 	if err != nil {
 		return nil, err
 	}
-	// TODO(monnand): With go 1.3, we cannot compress data now.
-	client.DisableCompression()
 
 	ret := &influxdbStorage{
 		client:         client,
 		machineName:    machineName,
-		tableName:      tablename,
+		database:       database,
 		bufferDuration: bufferDuration,
 		lastWrite:      time.Now(),
-		series:         make([]*influxdb.Series, 0),
+		points:         make([]*influxdb.Point, 0),
 	}
 	ret.readyToFlush = ret.defaultReadyToFlush
 	return ret, nil
 }
+
+// Creates a measurement point with a single value field
+func makePoint(name string, value interface{}) *influxdb.Point {
+	fields := map[string]interface{}{
+		fieldValue: toSignedIfUnsigned(value),
+	}
+
+	return &influxdb.Point{
+		Measurement: name,
+		Fields:      fields,
+	}
+}
+
+// Adds additional tags to the existing tags of a point
+func addTagsToPoint(point *influxdb.Point, tags map[string]string) {
+	if point.Tags == nil {
+		point.Tags = tags
+	} else {
+		for k, v := range tags {
+			point.Tags[k] = v
+		}
+	}
+}
+
+// Checks response for possible errors
+func checkResponseForErrors(response *influxdb.Response) error {
+	const msg = "failed to write stats to influxDb - %s"
+
+	if response != nil && response.Err != nil {
+		return fmt.Errorf(msg, response.Err)
+	}
+	if response != nil && response.Results != nil {
+		for _, result := range response.Results {
+			if result.Err != nil {
+				return fmt.Errorf(msg, result.Err)
+			}
+			if result.Series != nil {
+				for _, row := range result.Series {
+					if row.Err != nil {
+						return fmt.Errorf(msg, row.Err)
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// Some stats have type unsigned integer, but the InfluxDB client accepts only signed integers.
+func toSignedIfUnsigned(value interface{}) interface{} {
+	switch value.(type) {
+	case uint64:
+		if v, ok := value.(uint64); ok {
```

Comment:
Is the limitation of using just `int64` valid? Did you cross-verify against InfluxDB docs?